src/app/components/client/toolbar/UpsellBadge.tsx

```diff
@@ -155,44 +108,32 @@ function UpsellToggleButton(props: UpsellToggleButtonProps) {
   });
 
   return (
-    <>
-      <div className={styles.upsellWrapper}>
-        <button
-          {...buttonProps}
-          className={`${styles.upsellBadge} ${
-            state.isSelected ? styles.isSelected : ""
-          }`}
-          ref={ref}
-        >
-          {state.isSelected
-            ? l10n.getString("plus-indicator-label-active")
-            : l10n.getString("plus-indicator-label-inactive")}
-          <span className={styles.toggleIndicator} />
-        </button>
-        {props.experimentData?.["last-scan-date"].enabled &&
-          props.lastScanDate !== null && (
-            <span className={styles.lastScanIndicator}>
-              <Image src={LastScanIcon} alt="" width={31} height={25} />
-              <b>{l10n.getString("plus-indicator-scan-date-label")}</b>&nbsp;
-              {scanDateFormatter.format(props.lastScanDate)}
-            </span>
-          )}
-      </div>
-      {!props.enabledFeatureFlags.includes("SubscriptionPlansPage") &&
-        dialogState.isOpen && (
-          <UpsellDialog
-            {...overlayProps}
-            state={dialogState}
-            monthlySubscriptionUrl={props.monthlySubscriptionUrl}
-            yearlySubscriptionUrl={props.yearlySubscriptionUrl}
-            subscriptionBillingAmount={props.subscriptionBillingAmount}
-          />
+    <div className={styles.upsellWrapper}>
+      <UpsellLinkButton
+        {...buttonProps}
+        className={`${styles.upsellBadge} ${
+          state.isSelected ? styles.isSelected : ""
+        }`}
+        enabledFeatureFlags={props.enabledFeatureFlags}
+      >
+        {state.isSelected
+          ? l10n.getString("plus-indicator-label-active")
+          : l10n.getString("plus-indicator-label-inactive")}
+        <span className={styles.toggleIndicator} />
+      </UpsellLinkButton>
+      {props.experimentData?.["last-scan-date"].enabled &&
+        props.lastScanDate !== null && (
+          <span className={styles.lastScanIndicator}>
+            <Image src={LastScanIcon} alt="" width={31} height={25} />
+            <b>{l10n.getString("plus-indicator-scan-date-label")}</b>&nbsp;
+            {scanDateFormatter.format(props.lastScanDate)}
+          </span>
         )}
-    </>
+    </div>
   );
 }
 
-export type UpsellBadgeProps = UpsellButtonProps & {
+export type UpsellBadgeProps = UpsellLinkButtonProps & {
   lastScanDate: Date | null;
   enabledFeatureFlags: FeatureFlagName[];
```

Comment:
Maybe a little bit out of scope of this PR, but since it's such a small change... `enabledFeatureFlags` is already in `UpsellLinkButtonProps` :)